contracts/src/v0.8/functions/tests/1_0_0/FunctionsTermsOfServiceAllowList.t.sol

```diff
@@ -1,52 +1,343 @@
+contract FunctionsTermsOfServiceAllowList_GetConfig is FunctionsRoutesSetup {
+  function test_GetConfig_Success() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
 
+    TermsOfServiceAllowList.Config memory config = s_termsOfServiceAllowList.getConfig();
+    assertEq(config.enabled, getTermsOfServiceConfig().enabled);
+    assertEq(config.signerPublicKey, getTermsOfServiceConfig().signerPublicKey);
+  }
 }
 
 /// @notice #updateConfig
-contract FunctionsTermsOfServiceAllowList_UpdateConfig {
+contract FunctionsTermsOfServiceAllowList_UpdateConfig is FunctionsRoutesSetup {
+  function test_UpdateConfig_RevertIfNotOwner() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    vm.expectRevert("Only callable by owner");
+    s_termsOfServiceAllowList.updateConfig(
+      TermsOfServiceAllowList.Config({enabled: true, signerPublicKey: STRANGER_ADDRESS})
+    );
+  }
+
+  event ConfigUpdated(TermsOfServiceAllowList.Config config);
+
+  function test_UpdateConfig_Success() public {
+    TermsOfServiceAllowList.Config memory configToSet = TermsOfServiceAllowList.Config({
+      enabled: false,
+      signerPublicKey: TOS_SIGNER
+    });
+
+    // topic0 (function signature, always checked), NOT topic1 (false), NOT topic2 (false), NOT topic3 (false), and data (true).
+    bool checkTopic1 = false;
+    bool checkTopic2 = false;
+    bool checkTopic3 = false;
+    bool checkData = true;
+    vm.expectEmit(checkTopic1, checkTopic2, checkTopic3, checkData);
+    emit ConfigUpdated(configToSet);
+
+    s_termsOfServiceAllowList.updateConfig(configToSet);
 
+    TermsOfServiceAllowList.Config memory config = s_termsOfServiceAllowList.getConfig();
+    assertEq(config.enabled, configToSet.enabled);
+    assertEq(config.signerPublicKey, configToSet.signerPublicKey);
+  }
 }
 
 /// @notice #getMessage
-contract FunctionsTermsOfServiceAllowList_GetMessage {
+contract FunctionsTermsOfServiceAllowList_GetMessage is FunctionsRoutesSetup {
+  function test_GetMessage_Success() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
 
+    bytes32 message = s_termsOfServiceAllowList.getMessage(STRANGER_ADDRESS, STRANGER_ADDRESS);
+
+    assertEq(message, keccak256(abi.encodePacked(STRANGER_ADDRESS, STRANGER_ADDRESS)));
+  }
 }
 
 /// @notice #acceptTermsOfService
-contract FunctionsTermsOfServiceAllowList_AcceptTermsOfService {
+contract FunctionsTermsOfServiceAllowList_AcceptTermsOfService is FunctionsRoutesSetup {
+  function test_AcceptTermsOfService_RevertIfBlockedSender() public {
+    s_termsOfServiceAllowList.blockSender(STRANGER_ADDRESS);
+
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    bytes32 message = s_termsOfServiceAllowList.getMessage(STRANGER_ADDRESS, STRANGER_ADDRESS);
+    bytes32 prefixedMessage = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", message));
```

Comment:
put abi.encodePacked("\x19Ethereum Signed Message:\n32" in a contract-scoped var?